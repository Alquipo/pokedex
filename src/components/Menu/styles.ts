/* eslint-disable @typescript-eslint/no-non-null-assertion */
import styled, { css } from 'styled-components'
import { MenuProps } from '.'
import { ArrowBack } from '@styled-icons/ionicons-outline/ArrowBack'

export const wrapperModifiers = {
  left: () => css`
    display: flex;
    align-items: center;
    justify-content: flex-start;
  `,
  right: () => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    img {
      margin-right: 1rem;
    }
  `,
  center: () => css`
    display: flex;
    align-items: center;
    justify-content: center;
  `
}

export const BackButton = styled(ArrowBack)`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    width: 3rem;
    height: 3rem;
    cursor: pointer;
  `}
`

export const Wrapper = styled.header<MenuProps>`
  ${({ theme, bgColor, positionLogo }) => css`
    height: 6.4rem;
    width: 100%;
    overflow: hidden;
    padding: ${theme.spacings.xsmall} ${theme.spacings.small};
    margin-bottom: ${theme.spacings.xsmall};

    min-width: 450px;
    z-index: ${theme.layers.menu};

    position: sticky;
    top: 0;

    img {
      cursor: pointer;
    }

    background-color: ${theme.colors.pokemonText[bgColor!]};

    ${positionLogo && wrapperModifiers[positionLogo]}

    backdrop-filter: saturate(180%) blur(0.2rem);
    box-shadow: 0 0 40px 8px rgb(0 0 0 / 18%);
  `}
`
