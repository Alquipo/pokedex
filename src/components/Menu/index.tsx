import * as S from './styles'

const Menu = () => (
  <S.Wrapper>
    <h1>Menu</h1>
  </S.Wrapper>
)

export default Menu
