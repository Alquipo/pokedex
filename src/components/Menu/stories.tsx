import { Story, Meta } from '@storybook/react/types-6-0'
import Menu from '.'

export default {
  title: 'Reusable/Menu',
  component: Menu,
  args: {
    hasLogo: true,
    hasGitHubCorner: true,
    hasBackButton: true
  }
} as Meta

export const Default: Story = (args) => <Menu {...args} />
