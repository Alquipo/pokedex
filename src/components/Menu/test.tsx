import { render, screen } from '@testing-library/react'

import Menu from '.'

describe('<Menu />', () => {
  it('should render the heading', () => {
    const { container } = render(<Menu />)

    expect(screen.getByRole('heading', { name: /Menu/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
