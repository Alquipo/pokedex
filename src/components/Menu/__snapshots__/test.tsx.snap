// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render the correctly background-color menu 1`] = `
.c0 {
  height: 6.4rem;
  width: 100%;
  overflow: hidden;
  padding: 0.8rem 1.6rem;
  margin-bottom: 0.8rem;
  min-width: 450px;
  z-index: 20;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background-color: #ff8a80;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-backdrop-filter: saturate(180%) blur(0.2rem);
  backdrop-filter: saturate(180%) blur(0.2rem);
  box-shadow: 0 0 40px 8px rgb(0 0 0 / 18%);
}

.c0 img {
  cursor: pointer;
}

<header
  class="c0"
/>
`
